xclim/indices/_conversion.py

```diff
@@ -895,6 +909,115 @@ def rain_approximation(
     return prra
 
 
+@declare_units(snw="[mass]/[area]", snr="[mass]/[volume]")
+def snow_depth_approximation(
+    snw: xr.DataArray,
+    snr: xr.DataArray,
+) -> xr.DataArray:
+    """Approximation of snow depth from snow amount and density.
+
+    Parameters
+    ----------
+    snw : xr.DataArray
+        Snow amount [kg/m^2].
+        If snow water equivalent (`swe` [m]) is provided instead, will be converted to `snw` before calculating.
+    snr : xr.DataArray, optional
+        Snow density [kg/m^3].
+
+    Returns
+    -------
+    xr.DataArray, [m]
+        Snow depth.
+    """
+    snw = convert_units_to(snw, "kg m-2", context="hydro")
+    snr = convert_units_to(snr, "kg m-3")
+
+    snd = snw / snr
+
+    snd.attrs["units"] = "m"
+    return snd
+
+
+@declare_units(snd="[length]", snr="[mass]/[volume]")
+def snow_amount_approximation(
+    snd: xr.DataArray,
+    snr: xr.DataArray,
+) -> xr.DataArray:
+    """Approximation of snow amount from snow depth and density.
+
+    Parameters
+    ----------
+    snd : xr.DataArray
+        Snow depth [m].
+    snr : xr.DataArray
+        Snow density [kg/m^3].
+
+    Returns
+    -------
+    xr.DataArray, [kg m-2]
+        Surface snow amount
+    """
+    snd = convert_units_to(snd, "m", context="hydro")
```

Comment:
```suggestion
    snd = convert_units_to(snd, "m")
```
Same as above, but even more so : I see no case where the context would be useful.